src/pal/src/misc/sysinfo.cpp

```diff
@@ -102,7 +102,18 @@ PAL_GetLogicalCpuCountFromOS()
 {
     int nrcpus = 0;
 
-#if HAVE_SYSCONF
+#if HAVE_SCHED_GETAFFINITY
+
+    int pid = getpid();
+    cpu_set_t cpuSet;
+    int st = sched_getaffinity(pid, sizeof(cpu_set_t), &cpuSet);
```

Comment:
A nit -I've noticed the sched_getaffinity doc says that the pid argument can be zero to represent the calling process, so we don't need to call the getpid.